### PR TITLE
3 server setup first endpoint/task

### LINKED ISSUE #3: GET /documentation/:pageId/notes

### PR DESCRIPTION
**Pull request by andimiya**
- Sets up Serverless + Express!
- Routes are handled in `index.js`, it's a regular Express server
- Serverless will deploy the whole Express server
  - Use `sls deploy` to deploy to development server
- Uses `sls offline` for local development.
  - Run `sls offline start` in the terminal, which will spin up a local server at `localhost:3000` for local development

- Added endpoints below, which are successfully getting data from the database
  - GET `/documentation/:pageId/notes/:noteId`
  - GET `/documentation/:pageId/notes`

Closes #3 and #4

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const express = require("express");
+const db = require("./db");
+const bodyParser = require("body-parser");
+const app = express();
+
+app.use(express.static("public"));
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+
+app.get("/", function(req, res) {
```
Comment (brockinit) [inline index.js]: Let's remove this endpoint since it isn't part of the application.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const serverless = require("serverless-http");
+const express = require("express");
+const db = require("./db");
+const bodyParser = require("body-parser");
+const app = express();
+
+app.use(express.static("public"));
```
Comment (brockinit) [inline index.js]: We don't need this since we aren't rendering any pages.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+
+app.get("/", function(req, res) {
+  res.send("Hello World!");
+});
+
+app.get("/documentation/:pageId/notes", function(req, res) {
+  const query = "SELECT * FROM notes WHERE page_id = $1";
+  const page_id = req.params.pageId;
```
Comment (brockinit) [inline index.js]: Let's use camel casing here `pageId`.

File: index.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  const values = [page_id];
```
Comment (brockinit) [inline index.js]: Same here.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  if (!page_id) {
```
Comment (brockinit) [inline index.js]: Same here.

File: index.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    return res.status(400).json({ error: "Page Id required in the request" });
```
Comment (brockinit) [inline index.js]: This isn't necessary since this handler function won't be called if there isn't a `pageId`

File: serverless.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+service: awsWiki
```
Comment (brockinit) [inline serverless.yml:1]: Why do we need this file?

**Review by brockinit — CHANGES_REQUESTED** [review]
Looking good! Just need a few changes.

File: package.json [hunk 8]
```diff
@@ -20,11 +20,22 @@
...
   },
   "homepage": "https://github.com/brockinit/aws-wiki#readme",
   "dependencies": {
+    "body-parser": "1.18.3",
+    "express": "4.16.3",
     "husky": "^0.14.3",
-    "lint-staged": "^7.2.0"
+    "lint-staged": "^7.2.0",
+    "pg": "7.4.3",
+    "pg-hstore": "2.3.2",
```
Comment (brockinit) [inline package.json]: We can remove this.

File: index.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
+const serverless = require("serverless-http");
```
Comment (brockinit) [inline index.js:1]: This file should be within `serverless/`

File: index.js (hunk 1 above)
Comment (andimiya) [reply index.js]: Ah! Sorry, removed :)

File: index.js (hunk 2 above)
Comment (andimiya) [reply index.js]: 👍

File: index.js (hunk 3 above)
Comment (andimiya) [reply index.js]: Ok!

File: index.js (hunk 4 above)
Comment (andimiya) [reply index.js]: Updated! :)

File: index.js (hunk 5 above)
Comment (andimiya) [reply index.js]: 👍

File: serverless.yml (hunk 7 above)
Comment (andimiya) [reply serverless.yml:1]: The deletion didn't get pushed up, just pushed. Sorry :(

File: index.js (hunk 9 above)
Comment (andimiya) [reply index.js:1]: Serverless folder should've been removed but the changes didn't get pushed, can you check again please?

**Review by brockinit — APPROVED** [review]

**Pull request merged**